### PR TITLE
fix move & resize losing the grip when moving fast

### LINKED ISSUE #15: Move & Resize losing the grip when moving fast
**Problem:**
When moving or resizing a marker the delta is losing a fraction of a pixel while each `mousemove` event is triggering.
This leads to the mouse cursor not being at the same position as the grip after a while.

 **Solution:**
The solution is to replace the calculation of `dx` & `dy` variables in the `.move()` and `.resize()` functions with some exact computation depending on mouse `event.offsetX` & `event.offsetY`.

I successfully fixed this for the Rectangular and will open a pull request tomorrow after I also fix the line, arrow & text components.

### PR DESCRIPTION
**Pull request by philip-firstorder**
Fixes #15

### CODE REVIEW

File: src/markers/RectangularMarkerBase.ts [hunk 1]
```diff
@@ -41,65 +41,69 @@ export class RectangularMarkerBase extends MarkerBase {
         this.addControlBox();
     }
 
-    protected resize(x: number, y: number) {
-        let translateX = 0;
-        let translateY = 0;
+    protected resize(dx: number, dy: number) {
+        const width = this.previousState.width;
```
Comment (ailon) [inline src/markers/RectangularMarkerBase.ts]: `this.previousState` is undefined if resize is called before the mouse down (like in a state restore scenario)

**Comment by ailon** [comment]
Hey, thanks! There's an issue though. Now it doesn't work the second time around. 

To reproduce:
Try adding any rectangular marker, click the checkmark to render and then click on the image again.

This happens because you set `previousState` in the `mouseDown` event handler and it doesn't (and shouldn't) fire in the state restore scenario. I guess `previousState` should either be initialized to something or `undefined` checks should be present and handled wherever it is used.

**Comment by philip-firstorder** [comment]
I will check this and update the code in a couple of hours. Thanks for the check.

**Comment by philip-firstorder** [comment]
I added undefined checks because they are safer.

If undefined will initialise the constants to 0 so they won't add or subtract anything in subsequent operations.

**Comment by ailon** [comment]
thanks, Philip! Merged.

**Comment by ailon** [comment]
FYI, just noticed that it totally broke down in Firefox (all the manipulations are flipped). Investigating...

**Comment by ailon** [comment]
@philip-firstorder I decided to revert this for now as I can't see how to fix it quickly and I already managed to release the broken version.

Additionally, you chose to remove some things that I think were there for a reason (even if I can't remember them from the top of my head), like this one:
https://github.com/firstorder-gmbh/markerjs/commit/f7ff6fa2224f609e12540df3a779948cc3936816#diff-37bae59c6cc36a0a3f7d1c09c669c9d3L35

I just don't feel comfortable doing this without any specific justification.

**Comment by philip-firstorder** [comment]
Scale was only needed to convert the screenX, screenY pixels relative to your resized MarkerArea when you resize the window.

Since I used offsetX, offsetY of the scaled element it wasn't needed anymore.

Will check the firefox problem.

**Pull request merged**